File: cpp/unittests/SingleNumberTest.cpp
#include "SingleNumber.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(SingleNumber, Solution136_1) {
  std::vector<int> Nums1{2, 2, 1};
  const int Expected1 = 1;
  EXPECT_EQ(Expected1, Solution136_1().singleNumber(Nums1));

  std::vector<int> Nums2{4, 1, 2, 1, 2};
  const int Expected2 = 4;
  EXPECT_EQ(Expected2, Solution136_1().singleNumber(Nums2));
}

File: java/src/test/java/io/binac/leetcode/SingleNumberTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SingleNumberTest {
    private static final SingleNumber.Solution1 SOLUTION1 = new SingleNumber.Solution1();

    private final int[] NUMS1 = {2, 2, 1};
    private final int EXPECTED1 = 1;

    private final int[] NUMS2 = {4, 1, 2, 1, 2};
    private final int EXPECTED2 = 4;

    private final int[] NUMS3 = {1};
    private final int EXPECTED3 = 1;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.singleNumber(NUMS1));
        assertEquals(EXPECTED2, SOLUTION1.singleNumber(NUMS2));
        assertEquals(EXPECTED3, SOLUTION1.singleNumber(NUMS3));
    }
}